### PR TITLE
feat: limit persona and wallet name length

### PR DESCRIPTION
**Pull request by zhouhanseng**
Close https://github.com/DimensionDev/Maskbook/issues/1130#issuecomment-710355154

### CODE REVIEW

File: src/_locales/zh/messages.json [hunk 1]
```diff
@@ -266,6 +266,7 @@
     "wallets": "錢包",
     "untitled_wallet": "未命名錢包",
     "your_message": "您的留言……",
+    "input_length_exceed_prompt": "{{name}}的長度不應超過{{length}}。",
```
Comment (septs) [inline src/_locales/zh/messages.json:269]:
```suggestion
    "input_length_exceed_prompt": "{{name}}的長度不能超過{{length}}。",
```

@neruthes @Tedko

File: src/extension/options-page/DashboardDialogs/Persona.tsx [hunk 2]
```diff
@@ -63,7 +71,11 @@ export function DashboardPersonaCreateDialog(props: WrappedDialogProps) {
                     </>
                 }
                 footer={
-                    <DebounceButton type="submit" variant="contained" onClick={createPersonaAndNext} disabled={!name}>
+                    <DebounceButton
+                        type="submit"
+                        variant="contained"
+                        onClick={createPersonaAndNext}
+                        disabled={!name || name.length > 32}>
```
Comment (septs) [inline src/extension/options-page/DashboardDialogs/Persona.tsx:78]:
```
'💊'.length
2
Array.from('💊').length
1
```

File: src/extension/options-page/DashboardDialogs/Persona.tsx [hunk 3]
```diff
@@ -252,6 +264,14 @@ export function DashboardPersonaRenameDialog(props: WrappedDialogProps<PersonaPr
                 primary={t('persona_rename')}
                 content={
                     <TextField
+                        helperText={
+                            name.length > 32
+                                ? t('input_length_exceed_prompt', {
+                                      name: t('persona_name').toLowerCase(),
+                                      length: 32,
```
Comment (septs) [inline src/extension/options-page/DashboardDialogs/Persona.tsx:271]:
```suggestion
                                      length: 31,
```

File: src/_locales/zh/messages.json (hunk 1 above)
Comment (Tedko) [reply src/_locales/zh/messages.json:269]: Both is fine

File: src/extension/options-page/DashboardDialogs/Persona.tsx (hunk 3 above)
Comment (septs) [reply src/extension/options-page/DashboardDialogs/Persona.tsx:271]: condition is `> 32` not is `>= 32`

Comment (septs) [reply src/extension/options-page/DashboardDialogs/Persona.tsx:271]: extract to a function, (e.q: `isAvailableWalletName`)

File: src/extension/options-page/DashboardDialogs/Wallet.tsx [hunk 4]
```diff
@@ -345,9 +369,10 @@ export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
                         variant="contained"
                         onClick={onSubmit}
                         disabled={
-                            !(state[0] === 0 && name && confirmed) &&
-                            !(state[0] === 1 && name && mnemonic) &&
-                            !(state[0] === 2 && name && privKey)
+                            (!(state[0] === 0 && name && confirmed) &&
+                                !(state[0] === 1 && name && mnemonic) &&
+                                !(state[0] === 2 && name && privKey)) ||
+                            name.length > 32
```
Comment (septs) [inline src/extension/options-page/DashboardDialogs/Wallet.tsx:375]:
```suggestion
                            name.length > 32
```

extract to a function. (e.q: `isAvailableWalletName`)

File: src/extension/options-page/DashboardDialogs/Wallet.tsx [hunk 5]
```diff
@@ -261,6 +269,14 @@ export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
                 children: (
                     <div>
                         <TextField
+                            helperText={
+                                name.length > 32
+                                    ? t('input_length_exceed_prompt', {
+                                          name: t('wallet_name').toLowerCase(),
+                                          length: 32,
```
Comment (septs) [inline src/extension/options-page/DashboardDialogs/Wallet.tsx:276]:
```suggestion
                                          length: 32,
```

extract max length to constants file.

File: src/_locales/zh/messages.json (hunk 1 above)
Comment (septs) [reply src/_locales/zh/messages.json:269]: Waiting for @neruthes review the copywriting?

**Review by septs — APPROVED** [review]

File: src/utils/utils.ts [hunk 6]
```diff
@@ -240,3 +240,7 @@ export function assert(x: any, ...args: any): asserts x {
     console.assert(x, ...args)
     if (!x) throw new Error('Assert failed!')
 }
+
+export function checkInputLengthExceed(name: string) {
+    return Array.from(name).length > WALLET_OR_PERSONA_NAME_MAX_LEN
```
Comment (septs) [inline src/utils/utils.ts:245]:
```suggestion
    return name && Array.from(name).length > WALLET_OR_PERSONA_NAME_MAX_LEN
```

Comment (zhouhanseng) [reply src/utils/utils.ts]: done

Comment (zhouhanseng) [reply src/utils/utils.ts]: Why add this here?

Comment (septs) [reply src/utils/utils.ts:245]:
```
disabled={!name || checkInputLengthExceed(name)}
```

Comment (zhouhanseng) [reply src/utils/utils.ts:245]: 🤔I don't quite follow this, type of `name` is always string, maybe we can change this to：

```
disabled={name.length === 0 || checkInputLengthExceed(name)}
```

Comment (septs) [reply src/utils/utils.ts:245]: OK

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — DISMISSED** [review]

File: src/utils/utils.ts (hunk 6 above)
Comment (septs) [reply src/utils/utils.ts:245]:
```suggestion
    return Array.from(name).length >= WALLET_OR_PERSONA_NAME_MAX_LEN
```

i think, so `length: WALLET_OR_PERSONA_NAME_MAX_LEN - 1,` is doesn't need.

Comment (zhouhanseng) [reply src/utils/utils.ts]: done

**Review by septs — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Comment by septs** [comment]
@zhouhanseng plz resolve conflict.

**Comment by zhouhanseng** [comment]
solved

**Review by septs — APPROVED** [review]

**Pull request merged**